[Hub Generated] Review request for Microsoft.Security to add version stable/2019-08-01

liranc opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/liranc/azure-rest-api-specs/recommendations-2019-08-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for AME onboarding.
- [x] [Validation tools](https://aka.ms/swaggertools) have been run on swagger spec(s) updated and errors have all been fixed with this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, [please view the FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories.
*	adding/removing APIs.
* adding/removing properties.
* adding/removing API-version.
* adding a new service in Azure.

*Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.*

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
recos

Commit:
Remove system id

Commit:
Merge branch 'master' into recommendations-2019-08-01

AutorestCI commented:
/azp run automation - sdk

Commit:
spellcheck

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

Commit:
dataSource

AutorestCI commented:
/azp run automation - sdk

Commit:
Rename category to control

AutorestCI commented:
/azp run automation - sdk

Commit:
Merge branch 'master' into recommendations-2019-08-01

Commit:
remove provider, add couple of other fields

AutorestCI commented:
/azp run automation - sdk

Commit:
Merge branch 'master' into recommendations-2019-08-01

Commit:
productName/compnent

AutorestCI commented:
/azp run automation - sdk

majastrz commented on specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json:
```diff
@@ -208,6 +208,18 @@
           "example": "IoTSecurity",
           "description": "The name of the alert provider or internal partner"
         },
+        "productName": {
```
>productName [](start = 9, length = 11)

As per [https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md), evoluationary changes require a version bump.

majastrz commented on specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotRecommendations.json:
```diff
@@ -0,0 +1,237 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/iotRecommendations": {
```
>iotRecommendations [](start = 137, length = 18)

New APIs fall into a realm of evolutionary changes that require a version bump as per As per [https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md), evoluationary changes require a version bump..

Commit:
Merge branch 'master' into recommendations-2019-08-01

AutorestCI commented:
/azp run automation - sdk

Commit:
Merge branch 'master' into recommendations-2019-08-01

AutorestCI commented:
/azp run automation - sdk

pilor commented on specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotRecommendations.json:
```diff
@@ -0,0 +1,237 @@
+      "type": "object",
+      "description": "IoT recommendation",
+      "properties": {
+        "properties": {
```
this is missing the top level resource properties "type", "name", and "id".  Please inherit from the common "Resource" definition like iotRecommendationTypes does

adrianhall commented:
Per email discussion with Azure API Review Board - Exception from policy is granted, and changes are APPROVED without a version bump.

In the future, additive / evolutionary changes will require a version bump.

Commit:
Merge branch 'master' into recommendations-2019-08-01

Commit:
Inherit from resource, update example

AutorestCI commented:
/azp run automation - sdk

Commit:
Prettier

AutorestCI commented:
/azp run automation - sdk

liranc commented:
@pilor @adrianhall Thanks for reviewing! Could you please merge the PR?

pilor commented:
> @pilor @adrianhall Thanks for reviewing! Could you please merge the PR?

See step 8+ here:  https://armwiki.azurewebsites.net/rp_onboarding/process/api_review.html

liranc commented:
> See step 8+ here: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review.html

@pilor are there remaining comments/issues? As far as I understand steps 8+ are mostly about addressing feedback.
